pkg/background/update_request_controller.go

```diff
@@ -241,6 +245,59 @@ func (c *controller) syncUpdateRequest(key string) error {
 	return err
 }
 
+func (c *controller) checkIfCleanupRequired(ur *kyvernov1beta1.UpdateRequest) error {
+	var err error
+	pNamespace, pName, err := cache.SplitMetaNamespaceKey(ur.Spec.Policy)
+	if err != nil {
+		return err
+	}
+
+	if pNamespace == "" {
+		_, err = c.cpolLister.Get(pName)
+	} else {
+		_, err = c.polLister.Policies(pNamespace).Get(pName)
+	}
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		logger.V(4).Info("policy no longer exists, deleting the update request and respective resource based on synchronize", "ur", ur.Name, "policy", ur.Spec.Policy)
+		for _, e := range ur.Status.GeneratedResources {
+			if err := c.cleanupDataResource(e); err != nil {
+				logger.Error(err, "failed to clean up data resource on policy deletion")
+			}
+		}
+		return c.kyvernoClient.KyvernoV1beta1().UpdateRequests(config.KyvernoNamespace()).Delete(context.TODO(), ur.Name, metav1.DeleteOptions{})
```

Comment:
Not sure if we want to clean up URs on policy deletion, as discussed in the contributors meeting, would be good to think through and capture the UR lifecycle based on different settings.

Reply:
URs will not be deleted in case of clone policy with `sync enable as they will not be enqueued on policy delete.